httpserver/lmq_server.cpp

```diff
@@ -169,17 +170,90 @@ void LokimqServer::handle_notify_get_subscriber_count(
     lokimq_->send(message.conn, "COUNT", std::to_string(count));
 }
 
+bool LokimqServer::check_stats_key(const std::string& pk) const {
+
+    bool valid = false;
+
+    if (this->stats_access_key.empty()) {
+        LOKI_LOG(debug, "Access denied: stats-access-key is not specified");
+    } else if (pk != this->stats_access_key) {
+        LOKI_LOG(debug, "Access denied: unauthorised key");
+    } else {
+        valid = true;
+    }
+
+    return valid;
+}
+
+void LokimqServer::handle_get_logs(lokimq::Message& message) {
+
+    LOKI_LOG(debug, "Received get_logs request via LMQ");
+
+    if (!this->check_stats_key(message.conn.pubkey())) {
+        lokimq_->send(message.conn, "Access denied");
+        return;
+    }
+
+    /// Limit this call to 1 request per second
+    static time_t last_req_time = 0;
+    const time_t now = time(nullptr);
+    constexpr time_t PERIOD = 1;
+
+    const char* err_msg = nullptr;
+
+    if (now - last_req_time < PERIOD) {
+        err_msg = "Too many request, try again later.";
+    }
+
+    last_req_time = now;

```

Comment:
Since this endpoint is now access restricted, I think you could just drop this rate limiting.